byway/src/main/webapp/script.js

```diff
@@ -12,19 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- *
- */
-/* exported addRandomGreeting */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+function initMap() {
+  var directionsService = new google.maps.DirectionsService();
+  var directionsRenderer = new google.maps.DirectionsRenderer();
+  var start = new google.maps.LatLng(37.7699298, -122.4469157);
+  var end = new google.maps.LatLng(37.7683909618184, -122.51089453697205);
+  var mapOptions = {
+    zoom: 14,
+    center: start
+  }
+  var map = new google.maps.Map(document.getElementById('map'), mapOptions);
+  directionsRenderer.setMap(map);
+     document.getElementById("route").addEventListener("click", function() {
+    calcRoute(directionsService, directionsRenderer, start, end);
+  });
+}
+
+function calcRoute(directionsService, directionsRenderer, start, end) {
```

Comment:
Since javascript doesn't have typing, it would be useful to document what type of parameters you function expects. You can follow the style guide recommendations outlined here:
https://google.github.io/styleguide/jsguide.html#features-functions-top-level-functions

e.g. you could update this description like such:
```
/**
 * Calculates a route based on provided directions
 * @param {google.maps.DirectionsService} directionsService
 * @param {google.maps.DirectionsRenderer} directionsRenderer
 * @param {google.maps.LatLng} start
 * @param {google.maps.LatLng} end
 */
```

Reply:
Right, will do so.

Reply:
Acknowledged